### DISCUSSION TITLE
Add jpeg xl (.jxl) as another new image file format into J4

### DISCUSSION THREAD

**Discussion by Gitjk** [joomla/joomla-cms#34991] [2. Cold Features]
I would appreciate if the new jpeg xl (.jxl) image file format would be included in addition to the webp file format in J4 stable. Major browsers will soon support it, too.

See: https://jpegxl.info/

If I'm not mistaken, it would not require a code change, so '.jxl' could still be included into the J4 stable release.

**Comment by brianteeman** [comment]
Its a bit too early for me https://caniuse.com/?search=jpeg%20xl

Nothing stops anyone adding it to the list of supported formats on their own site although no one will be able to see the images yet

**Comment by dgrammatiko** [comment]
> although no one will be able to see the images yet

This is fixable, using a wasm decoder, eg from squoosh:

https://github.com/GoogleChromeLabs/squoosh/blob/dev/codecs/jxl/dec/jxl_dec.wasm

But I agree it's way too early

**Comment by richard67** [comment]
@Gitjk Would it be ok for you if we move this from issues to discussions? It's not really an issue as long as browsers don't support it, and you can already use that image format as described by @brianteeman above.

**Comment by Gitjk** [comment]
> 
> 
> @Gitjk Would it be ok for you if we move this from issues to discussions? It's not really an issue as long as browsers don't support it, and you can already use that image format as described by @brianteeman above.

No problem. My 'feature request' was just meant as a suggestion. Perhaps somebody with commit rights can add it on his ToDo list. :-)
I can display .jxl images already in the latest J4 RC with my copy of Firefox 92 nightly build.

**Comment by alikon** [comment]
converted to discussion

**Comment by richard67** [comment]
Would be good to have a timer bot which cyclically checks on https://caniuse.com/?search=jpeg%20xl and pings us if the result is green enough.
